runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -42,12 +61,14 @@ def generate_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
 
     initial = max(3 * horizon_timedelta, seasonality_timedelta)
 
+    print(f"{initial}, {horizon}, {unit}, {num_folds}")
+
     # Last cutoff is "latest date in data - horizon_timedelta" date
     cutoff = df["ds"].max() - horizon_timedelta
     if cutoff < df["ds"].min():
         raise ValueError("Less data than horizon.")
     result = [cutoff]
-    while result[-1] >= min(df["ds"]) + initial and len(result) < num_folds:
+    while result[-1] >= min(df["ds"]) + initial and len(result) <= num_folds:
```

Comment:
I think this is a bug. In `utils_test.py`, unit tests for `generate_cutoffs` were producing one less fold than the `num_folds` argument.

Reply:
Ahh I think you're right. The previous way might have counted the initial period for the `num_folds`, so the actual number of validation sets is at most `num_folds - 1`